src/styles/index.scss

```diff
@@ -1,3 +1,71 @@
-body {
+$color-block: #eee;
+$start-days: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+$month-lengths: (28, 29, 30, 31);
+$calendar-width: 706;
+
+.page {
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
   margin: 0;
 }
+
+.calendar {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  cursor: pointer;
+  display: flex;
+  flex-flow: row wrap;
+  margin: 81.5px 151px;
+  gap: 1px;
+  width: $calendar-width;
+  justify-content: flex-start;
+  list-style: none;
+  padding: 0;
+
+  &__day {
+    width: 100px;
+    height: 100px;
+    text-align: center;
+    line-height: 98px;
+    border: 1px solid black;
+    background-color: $color-block;
+    transition-duration: 0.5s;
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &__day:hover {
+    background-color: #ffbfcb;
+    transform: translate(0, -20px);
+  }
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+    }
+  }
```

Comment:
While using `nth-child` is a clever way to add content, it can lead to maintenance issues if the structure of the calendar changes. Consider if there's a more maintainable way to handle this, such as generating the days with HTML and controlling the display with SCSS.